packages/@uppy/companion/src/companion.js

```diff
@@ -73,6 +73,7 @@ module.exports.app = (optionsArg = {}) => {
   const providers = providerManager.getDefaultProviders()
 
   providerManager.addProviderOptions(options, grantConfig)
+  options.providerOptions = merge(options.providerOptions, grantConfig)
```

Comment:
not sure about this change

Reply:
This is here because `grant` does interpolation for certain settings passed as GET parameters to the connect handler if they are marked as dynamic, e.g. if you add subdomain to the dynamic array, you can use it in authorize_url and access_url (c.f. https://github.com/simov/grant#dynamic-state and    https://github.com/owncloud/uppy/blob/3403194fd4692c679e03d993c040c6d8ac5ac29a/packages/%40uppy/companion/src/config/grant.js#L47-L57 - yes, it should only be used as a subdomain not as a fqn but that's a dlfferent topic). 

To make this work in companion we need to forward the dynamic paramaters in the redirect from our connect handler to the grant connect handler. 
In order to achieve that, I merge the grant config including the dynamic option back to providerOptions and loop over the `dynamic` array in our connect handler (which is the other part of the code you commented on)

Reply:
ffr https://github.com/owncloud/uppy/blob/3403194fd4692c679e03d993c040c6d8ac5ac29a/packages/%40uppy/companion/src/server/provider/webdav/auth/index.js#L30